modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotAcceptanceManager.scala

```diff
@@ -0,0 +1,141 @@
+package org.tessellation.infrastructure.snapshot
+
+import cats.data.NonEmptyList
+import cats.effect.Async
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.traverse._
+
+import scala.collection.immutable.{SortedMap, SortedSet}
+
+import org.tessellation.ext.crypto._
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema._
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.{Amount, Balance}
+import org.tessellation.schema.block.DAGBlock
+import org.tessellation.schema.transaction.{DAGTransaction, RewardTransaction}
+import org.tessellation.sdk.domain.block.processing._
+import org.tessellation.security.signature.Signed
+import org.tessellation.statechannel.StateChannelSnapshotBinary
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.NonNegLong
+
+trait GlobalSnapshotAcceptanceManager[F[_]] {
+  def accept(
+    blocksForAcceptance: List[Signed[DAGBlock]],
+    scEvents: List[StateChannelEvent],
+    rewards: SortedSet[RewardTransaction],
+    lastSnapshotContext: GlobalSnapshotContext,
+    lastActiveTips: SortedSet[ActiveTip],
+    lastDeprecatedTips: SortedSet[DeprecatedTip]
+  ): F[
+    (
+      BlockAcceptanceResult[DAGBlock],
+      SortedMap[Address, NonEmptyList[Signed[StateChannelSnapshotBinary]]],
+      Set[StateChannelEvent],
+      SortedSet[RewardTransaction],
+      GlobalSnapshotInfo
+    )
+  ]
+}
```

Comment:
@tbekas can I fix this comment in the next PR? I'm going to create a ticket related to incremental snapshots and put there all the things that we won't fix currently

Reply:
I think it may be addressed by @kriscieslak in response to this comment https://github.com/Constellation-Labs/tessellation/pull/577#discussion_r1143636577

Anyway, I'm fine with fixing it later on.